views/js/pciCreator/dev/audioRecordingInteraction/runtime/audioRecordingInteraction.js

```diff
@@ -0,0 +1,720 @@
+
+            $micIcon.empty();
+            $micIcon.append($('<img>',
+                { src: this.assetManager.resolve(ICON_MIC)}
+            ));
+
+            this.inputMeter = uiElements.inputMeterFactory({
+                $container: this.$meterContainer.find('.leds'),
+                maxLevel: 100 // this is closely related to the values analyser.minDecibels and analyser.maxDecibels in recorderFactory
+            });
+        },
+
+        /**
+         * Instanciate the media stimulus player and its event listeners
+         * This player is only for the playback of the stimulus. The recorded audio uses its own player.
+         */
+        initMediaStimulus: function initMediaStimulus() {
+            var self = this;
+
+            if (this.hasMediaStimulus()) {
+                this.$mediaStimulusContainer.addClass('active');
+
+                this.mediaStimulus = uiElements.mediaStimulusFactory({
+                    $container:   this.$mediaStimulusContainer,
+                    assetManager: this.assetManager,
+                    media:        this.config.media
+                });
+
+                this.mediaStimulus.on('statechange', function() {
+                    self.updateControls();
+                });
+
+                this.mediaStimulus.on('playing', function() {
+                    if (self.recorder.is('recording')) {
+                        self.recorder.cancel();
+                    }
+                    if (self.player.is('playing')) {
+                        self.player.stop();
+                    }
+                });
+
+                this.mediaStimulus.on('ended', function() {
+                    if (self.config.autoStart) {
+                        self.startRecording();
+                    }
+                });
+                this.mediaStimulus.render();
+
+            } else {
+                this.$mediaStimulusContainer.empty();
+                this.$mediaStimulusContainer.removeClass('active');
+            }
+        },
+
+        /**
+         * Check if the item has a media stimulus defined
+         * @returns {Boolean}
+         */
+        hasMediaStimulus: function hasMediaStimulus() {
+            return (this.config.useMediaStimulus && this.config.media && this.config.media.uri);
+        },
+
+        /**
+         * Check if the media stimulus has been played
+         * @returns {Boolean}
+         */
+        mediaStimulusHasPlayed: function mediaStimulusHasPlayed() {
+            return this.mediaStimulus && (this.mediaStimulus.is('ended') || this.mediaStimulus.is('disabled'));
+        },
+
+        /**
+         * Starts the recording if has permission to access the mic. If not, ask for it.
+         */
+        startRecording: function startRecording() {
+            var self = this;
+
+            if (this.recorder.is('created')) {
+                this.recorder.init().then(function() {
+                    startForReal();
+                });
```

Comment:
What happens if the promise is rejected ? there is no error catch. Is is wanted ?